### PR TITLE
test(client): add batch put test for duplicate keys

### PR DESCRIPTION
**Pull request by xiaguan**
Removed the CHECK on master to avoid a crash, and added a test to cover this case.

### CODE REVIEW

File: mooncake-store/src/client.cpp [hunk 1]
```diff
@@ -620,7 +623,7 @@ void Client::StartBatchPut(std::vector<PutOperation>& ops,
         return;
     }
 
-    // Process individual responses with robust error handling
+    // Process individual responses with robust error handlingw
```
Comment (ykwd) [inline mooncake-store/src/client.cpp]: typeo

Comment (xiaguan) [reply mooncake-store/src/client.cpp]: That's odd—the CI didn't catch the typos.

Thanks for pointing them out! I'll get them fixed.

**Review by ykwd — APPROVED** [review]
